### PR TITLE
Add test to make sure option names won't change

### PR DESCRIPTION
**Pull request by Cosifne**
Once an option is added to the LSP server, its name and options group should not be modified.
I talked to @333fred during lunchtime and he said it doesn't meet the public API bar so these don't need to go via the API review.

Add a test to guard this

### CODE REVIEW

File: src/Features/LanguageServer/Protocol/Handler/Configuration/DidChangeConfigurationNotificationHandler_OptionList.cs [hunk 1]
```diff
@@ -26,7 +26,6 @@ internal partial class DidChangeConfigurationNotificationHandler
             ImplementTypeOptionsStorage.PropertyGenerationBehavior,
             // Completion
             CompletionOptionsStorage.ShowNameSuggestions,
-            CompletionOptionsStorage.ShowItemsFromUnimportedNamespaces,
```
Comment (Cosifne) [inline src/Features/LanguageServer/Protocol/Handler/Configuration/DidChangeConfigurationNotificationHandler_OptionList.cs:29]: This is a duplicate (See line 31)

**Review by dibarbet — APPROVED** [review]

File: src/Features/LanguageServer/ProtocolUnitTests/Configuration/DidChangeConfigurationNotificationHandlerTest.cs [hunk 2]
```diff
@@ -104,6 +105,48 @@ public class A { }";
...
 
+        [Fact]
+        public void VerifyLspClientOptionNames()
+        {
+            var actualNames = DidChangeConfigurationNotificationHandler.SupportedOptions.Select(
+                DidChangeConfigurationNotificationHandler.GenerateFullNameForOption).ToArray();
+            // These options are persist in the LSP client. Please make sure also modify the LSP client code if these strings are changed.
+            var expectedNames = new[]
+            {
+                "symbol_search.dotnet_search_reference_assemblies",
```
Comment (jasonmalinowski) [inline src/Features/LanguageServer/ProtocolUnitTests/Configuration/DidChangeConfigurationNotificationHandlerTest.cs:116]: Should we have this sorted and use one of our sort-ignoring Assert functions?

File: src/Features/LanguageServer/ProtocolUnitTests/Configuration/DidChangeConfigurationNotificationHandlerTest.cs [hunk 3]
```diff
@@ -104,6 +105,48 @@ public class A { }";
...
+                "implement_type.dotnet_insertion_behavior",
+                "implement_type.dotnet_property_generation_behavior",
+                "completion.dotnet_show_name_completion_suggestions",
+                "completion.dotnet_provide_regex_completions",
```
Comment (jasonmalinowski) [inline src/Features/LanguageServer/ProtocolUnitTests/Configuration/DidChangeConfigurationNotificationHandlerTest.cs:120]: Shouldn't this be something like dotnet.completion.provideRegexCompletions? Or am I misunderstanding what this test is trying to do?

File: src/Features/LanguageServer/ProtocolUnitTests/Configuration/DidChangeConfigurationNotificationHandlerTest.cs (hunk 2 above)
Comment (Cosifne) [reply src/Features/LanguageServer/ProtocolUnitTests/Configuration/DidChangeConfigurationNotificationHandlerTest.cs:116]: Yeah it's possible

**Pull request merged**